src/frontend/servers/y-provider/__tests__/content.test.ts

```diff
@@ -0,0 +1,207 @@
+  });
+
+  test('POST /api/content with missing format parameter', async () => {
+    const app = initApp();
+
+    const response = await request(app)
+      .post('/api/content')
+      .set('origin', origin)
+      .set('authorization', apiKey)
+      .set('content-type', 'application/json')
+      .send({
+        content: 'dGVzdA==',
+      });
+
+    expect(response.status).toBe(400);
+    expect(response.body).toStrictEqual({
+      error: 'Invalid format. Must be one of: json, markdown, html',
+    });
+  });
+
+  test('POST /api/content with invalid format', async () => {
+    const app = initApp();
+
+    const response = await request(app)
+      .post('/api/content')
+      .set('origin', origin)
+      .set('authorization', apiKey)
+      .set('content-type', 'application/json')
+      .send({
+        content: 'dGVzdA==',
+        format: 'invalid',
+      });
+
+    expect(response.status).toBe(400);
+    expect(response.body).toStrictEqual({
+      error: 'Invalid format. Must be one of: json, markdown, html',
+    });
+  });
+
+  test.each([
+    { authHeader: `Bearer ${apiKey}`, format: 'json' },
+    { authHeader: `Bearer ${apiKey}`, format: 'markdown' },
+    { authHeader: `Bearer ${apiKey}`, format: 'html' },
+  ])(
+    'POST /api/content with correct content and format $format with Authorization: $authHeader',
+    async ({ authHeader, format }) => {
+      const app = initApp();
+
+      // Create a simple Yjs document for testing using BlockNote
+      const editor = ServerBlockNoteEditor.create();
+      const markdownContent = '# Test Document\n\nThis is test content.';
+      const blocks = await editor.tryParseMarkdownToBlocks(markdownContent);
+      const yDocument = editor.blocksToYDoc(blocks, 'document-store');
+      const yjsUpdate = Y.encodeStateAsUpdate(yDocument);
+      const base64Content = Buffer.from(yjsUpdate).toString('base64');
+
+      const response = await request(app)
+        .post('/api/content')
+        .set('Origin', origin)
+        .set('Authorization', authHeader)
+        .set('content-type', 'application/json')
+        .send({
+          content: base64Content,
+          format: format,
+        });
+      expect(response.status).toBe(200);
+      expect(response.body).toHaveProperty('content');
+      expect(response.body).toHaveProperty('format', format);
+
+      // Verify the content based on format
+      if (format === 'json') {
+        const parsedContent = response.body.content;
+        expect(Array.isArray(parsedContent)).toBe(true);
+        expect(parsedContent.length).toBe(2);
+        expect(parsedContent[0].type).toBe('heading');
+        expect(parsedContent[1].type).toBe('paragraph');
+        expect(parsedContent[0].content[0].type).toBe('text');
+        expect(parsedContent[0].content[0].text).toBe('Test Document');
+        expect(parsedContent[1].content[0].type).toBe('text');
+        expect(parsedContent[1].content[0].text).toBe('This is test content.');
```

Comment:
I'd do a `expect(...).toStrictEqual()`. You are not testing the output fully, constraining what comes out of it beyond the specific fields you are testing, etc.